api/api/controllers/search_controller.py

```diff
@@ -166,16 +165,16 @@ def _post_process_results(
     return results[:page_size]
 
 
-def get_excluded_providers_query() -> Q | None:
+def get_filtered_providers_query() -> Q | None:
     """
-    Hide data sources from the catalog dynamically.
+    Get a query for filtering out items from hidden content providers or without any content provider.
```

Comment:
As a supplement to my other comment, this would be clearer, maybe with an optional additional line that clarifies how we decide if a provider is "enabled", if the code is for some reason insufficient.

```suggestion
    Get a query that only includes enabled providers.
```